packages/aws-cdk-lib/core/lib/private/synthesis.ts

```diff
@@ -254,6 +261,90 @@ function invokeAspects(root: IConstruct) {
   }
 }
 
+/**
+ * Invoke aspects V2 runs a stabilization loop and allows Aspects to invoke other Aspects.
+ * This function also does not emit a warning for nested Aspects. Runs if the feature flag
+ * '@aws-cdk/core:aspectStabilization' is enabled.
+ */
+function invokeAspectsV2(root: IConstruct) {
+  const invokedByPath: { [nodePath: string]: AspectApplication[] } = { };
+
+  recurse(root, []);
+
+  for (let i = 0; i <= 100; i++) {
+    const didAnythingToTree = recurse(root, []);
+
+    if (!didAnythingToTree) {
+      return;
+    }
+  }
+
+  throw new Error('Maximum iterations reached while invoking Aspects.');
+
+  function recurse(construct: IConstruct, inheritedAspects: AspectApplication[]): boolean {
+    const node = construct.node;
+    const aspects = Aspects.of(construct);
+
+    let didSomething = false;
+
+    let localAspects = aspects.list;
+    const allAspectsHere = sortAspectsByPriority(inheritedAspects, localAspects);
+
+    for (const aspectApplication of allAspectsHere) {
+
+      let invoked = invokedByPath[node.path];
+      if (!invoked) {
+        invoked = invokedByPath[node.path] = [];
+      }
+
+      if (invoked.includes(aspectApplication)) { continue; }
+
+      // If the last invoked Aspect has a higher priority than the current one, throw an error:
+      const lastInvokedAspect = invoked[invoked.length - 1];
+      if (lastInvokedAspect && lastInvokedAspect.priority > aspectApplication.priority) {
+        throw new Error(
+          `Aspect ${aspectApplication.aspect.constructor.name} with priority ${aspectApplication.priority} invoked after ${lastInvokedAspect.aspect.constructor.name} with priority ${lastInvokedAspect.priority} at ${node.path}.`,
```

Comment:
Was it invoked? Not really right? Also, what's the user-visible behavior that triggers this?

Isn't it more like "aspect with a low priority got added by an aspect with a high priority", or something like that?